src/machine.rs

```diff
@@ -1022,7 +1028,12 @@ impl OlmMachine {
                 .import_backed_up_room_keys(keys, |progress, total| {
                     if let Some(callback) = &progress_listener {
                         callback
-                            .call2(&JsValue::NULL, &JsValue::from(progress), &JsValue::from(total))
+                            .call3(
+                                &JsValue::NULL,
+                                &JsValue::from(progress),
+                                &JsValue::from(total + failures),
```

Comment:
I don't understand this line. Does the total not include failures too? If so, should the `total` variable be renamed `successes` maybe?

Reply:
You're right, it's not very clear.  I've added a comment and renamed the variable, so it should be clearer now.